fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalViewScanOperator.java

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.operator.logical;
+
+import com.starrocks.analysis.Expr;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.Table;
+import com.starrocks.sql.optimizer.base.ColumnRefSet;
+import com.starrocks.sql.optimizer.operator.Operator;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.OperatorVisitor;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+
+import java.util.Map;
+import java.util.Objects;
+
+// the logical operator to scan view just like LogicalOlapScanOperator to scan olap table,
+// which is a virtual logical operator used by view based mv rewrite and has no corresponding physical operator.
+// So the final plan will never contain an operator of this type.
+public class LogicalViewScanOperator  extends LogicalScanOperator {
```

Comment:
LogicalViewScanOperator  can reference to CTEConsumerOperator, we can put the substitute OptExpression just inside of LogicalViewScanOperator as a field instead of using a HashMap to keep their relationship.

Reply:
I will add a field OptExpression originalPlan in LogicalViewScanOperator.